marge.py

```diff
@@ -1,53 +1,176 @@
+
+
+def source_providers() -> list[SourceProvider]:
     datestr = "2012-09-19T12:43:00.000+00:00"
     return [
         SourceProvider(id=i[0], created_on=datestr, name=i[1], updated_on=datestr)
-        for i in enumerate(providers, start=1)
+        for i in enumerate(PROVIDERS, start=1)
     ]
+
+
+def configured_sources(
+    settings: Settings, account: str, device: str
+) -> list[ConfiguredSource]:
+    sources_tree = ET.parse(
+        account_device_dir(settings, account, device) + "Sources.xml"
+    )
+    root = sources_tree.getroot()
+    sources_list = []
+    for source_elem in root.findall("source"):
+        display_name = source_elem.attrib.get("displayName", "")
+        secret = source_elem.attrib.get("secret", "")
+        secret_type = source_elem.attrib.get("secretType", "")
+        source_key_elem = source_elem.find("sourceKey")
+        source_key_account = source_key_elem.attrib.get("account", "")
+        source_key_type = source_key_elem.attrib.get("type", "")
+        sources_list.append(
+            ConfiguredSource(
+                display_name=display_name,
+                secret=secret,
+                secret_type=secret_type,
+                source_key_type=source_key_type,
+                source_key_account=source_key_account,
+            )
+        )
+
+    return sources_list
+
+
+def presets(settings: Settings, account: str, device: str) -> list[Preset]:
+    storedTree = ET.parse(account_device_dir(settings, account, device) + "Presets.xml")
+    root = storedTree.getroot()
+
+    presets = []
+
+    for preset in root.findall("preset"):
+        id = preset.attrib["id"]
+        content_item = preset.find("ContentItem")
+        name = content_item.find("itemName").text
+        source = content_item.attrib["source"]
+        type = content_item.attrib["type"] if "type" in content_item.attrib else ""
+        location = content_item.attrib["location"]
+        source_account = content_item.attrib["sourceAccount"]
+        is_presetable = content_item.attrib["isPresetable"]
+        container_art_elem = content_item.find("containerArt")
+        if container_art_elem != None and container_art_elem.text != None:
+            container_art = container_art_elem.text
+        else:
+            container_art = ""
+
+        presets.append(
+            Preset(
+                name=name,
+                id=id,
+                source=source,
+                type=type,
+                location=location,
+                source_account=source_account,
+                is_presetable=is_presetable,
+                container_art=container_art,
+            )
+        )
+
+    return presets
+
+
+def presets_xml(settings: Settings, account: str, device: str) -> ET.Element:
+    conf_sources_list = configured_sources(settings, account, device)
+
+    presets_list = presets(settings, account, device)
+
+    datestr = "2012-09-19T12:43:00.000+00:00"
```

Comment:
```suggestion
    # We hardcode a date here because we'll never use it, so there's no need for a real date object.
    datestr = "2012-09-19T12:43:00.000+00:00"
```